### ISSUE TITLE
[Buttons] Dynamic type does update titleFont:forState:

### ISSUE THREAD

**Issue by codeman7** [material-components/material-components-ios#7492]
Enabling `mdc_adjustFontForContentSizeCategory` updates the `titleLabel.font` on an MDCButton but does not update the `_font` dictionary. This results in `titleFont:forState:` returning a value that's different than `titleLabel.font`.

#### Repro steps

```swift
let button = MDCButton()
button.setTitleFont(customFont, for: .normal)
button.mdc_adjustFontForContentSizeCategory = true
```

#### Actual behavior

```swift
button.titleLabel.font != button.titleFont(for: .normal)
```

#### Expected behavior

```swift
button.titleLabel.font != button.titleFont(for: .normal)
```

---
#### Internal data
- Associated internal bug: [b/133514005](http://b/133514005)

**Comment by jverkoey** [comment]
The actual and expected behaviors here look the same, and the issue title looks like it may not be describing a problem. Please clarify what the bug is here.

**Comment by codeman7** [comment]
This was fixed in #7514
